lib/puppet/parser/functions/zookeeper_servers_url.rb

```diff
@@ -0,0 +1,27 @@
+module Puppet::Parser::Functions
+  newfunction(:zookeeper_servers_url, :type => :rvalue, :doc => <<-EOS
+This function converts an array of zookeeper hostnames into a combined URL for
+zookeeper HA. You may pass a second argument to indicate the zookeeper port.
```

Comment:
> You may pass a second argument to indicate the zookeeper port.

I don't see where would you parse the port variable. I guess it's much easier to leave the URI as an array of `ipaddress[:port]`.